### ISSUE TITLE
terminate called after throwing an instance of 'optix::Exception'

### ISSUE THREAD

**Issue by micaeltchapmi** [NVIDIA/VisRTX#18]
Hi, 

Thanks for this great Rendering Framework. I modified SampleMDLHDRI.cpp to render several objects in a 3D Scene with different materials and I successfully rendered a few objects but when I attempt to render all objects(~100objects), I get the error message below: 

_terminate called after throwing an instance of 'optix::Exception'
 what():  Unknown error (Details: Function "RTresult _rtContextLaunch2D(RTcontext, unsigned int, RTsize, RTsize)" caught exception: Encountered a rtcore error: m_api.pipelineCreate( context, pipelineOptions, compileOptions, modules, moduleCount, pipeline ) returned (5): Out of constant space)_

I'm not so sure what the error message means. From the message, I think it is a space issue but I monitored my CPU and GPU memory during the rendering and I always had a lot of free space on both devices. Can you help me understand what this error means?

Thanks,

Micael

**Comment by tbiedert** [comment]
Hi Micael, thanks for reaching out!

This is most likely an internal VisRTX or OptiX issue.

Could you provide a minimal reproducer?  

Also, please describe your setup (driver, OptiX version, MDL version, ..).

Thanks!
Tim

**Comment by micaeltchapmi** [comment]
Hi Tim,

Thanks for your reply. Here is my setup:

OS: Ubuntu 16.04
Optix: NVIDIA-OptiX-SDK-6.0.0-linux64 
MDL: MDL-SDK-2019 
CUDA Version: 10.2
Driver Version: 440.59

[https://drive.google.com/open?id=17iky6disTF7Tgv5NHuOPbBftNwFrGUO0](url) 
The above link contains a minimal setup to reproduce the error. The script is in test.zip and to run it, I use:

./visRtxSampleMDLHDRI pathto/allobjs pathto/nvidia/vMaterials/ pathto/shortlabels.txt

I included the shortlabels.txt file and allobjs file which contains all the objects I am trying to render. Let me know if you need anything else.

Thanks,

Micael

**Comment by tbiedert** [comment]
Thanks! We'll try to reproduce the issue.

In the meantime, could you please try with latest OptiX 6.5 and MDL SDL 2020?

I typically build and run like this:
```
•	cmake -GNinja -DOptiX_INSTALL_DIR=/opt/NVIDIA-OptiX-SDK-6.5.0-linux64 -DMDL_INSTALL_DIR=/opt/mdl-sdk-327300.3640 ..
•	ninja
•	LD_LIBRARY_PATH=/opt/NVIDIA-OptiX-SDK-6.5.0-linux64/lib64:/opt/mdl-sdk-327300.3640/linux-x86-64/lib ./visRtxSampleMDL
```

**Comment by micaeltchapmi** [comment]
Hi, 

Thanks for your suggestion. I will look into that. 

Given that some of my objects have the same material, I noticed that I was loading the same material several times and to solve this, I saved materials that were already loaded into a dictionary(map) and loaded new materials only if they were not already loaded. So far, it has helped solve the issue but I'm not sure what will happen with many more different materials.
